### ISSUE TITLE
Gradual training strategy

### ISSUE THREAD

**Issue by kartik4949** [google/automl#798]
Gradual training strategy i.e training network for few epochs with all layers freezed , 
next resume training with unfreezing some last layers , finally training all layers for finetuning.
can this be helpfull for our network for custom dataset.
@mingxingtan @fsx950223

**Comment by mingxingtan** [comment]
Yeah, sounds like a good (and simple optimization) to speed up the training, but I am not sure about the quality impact.

**Comment by kartik4949** [comment]
@mingxingtan quality impact should be slim to none. 
at least on theory we can try atleast.

**Comment by ChulanZhang** [comment]
@kartik4949 
I am a beginner at ML. May I know the reason for the sequence of your idea? Because I am having an opposite idea of sequence for training. Feel free to correct me if I am wrong.

Intuitively, I think that if I need to finetune the model on a custom dataset:
I should first train the whole network without freeze any part of it. This is to make the model get some features of the new classes in the custom dataset.
After achieving a decent number, I should freeze part of the network, for example, the 'efficientnet' backbone part. 
And then continue the training to the end.

I can find "var_freeze_expr: '(efficientnet|fpn_cells|resample_p6)'" in the tutorial.
And may I know all the available options for 'var_freeze_expr'?
